specifyweb/frontend/js_src/lib/components/Molecules/AttachmentPicker.tsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+
+import { useBooleanState } from '../../hooks/useBooleanState';
+import { useErrorContext } from '../../hooks/useErrorContext';
+import { attachmentsText } from '../../localization/attachments';
+import { commonText } from '../../localization/common';
+import { preferencesText } from '../../localization/preferences';
+import { wbText } from '../../localization/workbench';
+import { Tabs } from '../AppResources/Tabs';
+import { Button } from '../Atoms/Button';
+import { Textarea } from '../Atoms/Form';
+import { AttachmentsView } from '../Attachments';
+import { fetchOriginalUrl } from '../Attachments/attachments';
+import { UploadAttachment, useAttachment } from '../Attachments/Plugin';
+import { LoadingContext } from '../Core/Contexts';
+import { Dialog } from './Dialog';
+
+const types = ['url', 'image', 'attachments', 'attachments'] as const;
+
+export function AttachmentPicker({
+  url,
+  onChange: handleChange,
+  isReadOnly,
+}: {
+  readonly url: string | undefined;
+  readonly onChange: (url: string) => void;
+  readonly isReadOnly: boolean;
+}): JSX.Element {
+  const [attachment, setAttachment] = useAttachment(undefined);
+  useErrorContext('attachment', attachment);
+
+  const [isOpen, , , handleToggle] = useBooleanState();
```

Comment:
Currently, handleToggle isn't getting called when you click on an attachment. Would be better if dialog also closes when user clicks on an attachment.

Reply:
Instead of choosing an attachment preview than closing dialog?

Reply:
Good catch. The `onUploaded` callback for `<UploadAttachment>` and `onClick` callback for `<AttachmentsView>` should both call `handleClose` to close the dialog afterward